src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -318,22 +319,104 @@ private static boolean isAnonInnerClassInit(DetailAST lcurly) {
     }
 
     /**
-     * Collects validation details.
-     * @param ast detail ast.
-     * @return object that contain all details to make a validation.
+     * Checks if definition body is empty.
+     * @param lcurly left curly.
+     * @return true if definition body is empty.
+     */
+    private static boolean isEmptyBody(DetailAST lcurly) {
+        boolean result = false;
+        if (lcurly.getParent().getType() == TokenTypes.OBJBLOCK) {
+            if (lcurly.getNextSibling().getType() == TokenTypes.RCURLY) {
+                result = true;
+            }
+        }
+        else if (lcurly.getFirstChild().getType() == TokenTypes.RCURLY) {
+            result = true;
+        }
+        return result;
+    }
+
+    /**
+     * Finds next token after the given one.
+     * @param ast the given node.
+     * @return the token which represents next lexical item.
+     */
+    private static DetailAST getNextToken(DetailAST ast) {
```

Comment:
Shouldn't this be a method of `Details`? It is only used there.
If so, move any other methods that only `Details` uses to inside it.